src/web/components/App.tsx

```diff
@@ -386,6 +388,27 @@ export const App = ({ CAPI, NAV }: Props) => {
 					/>
 				</HydrateOnce>
 			))}
+			{CAPI.quizAtoms.map((quizAtoms, index) => (
+				<HydrateOnce
+					key={index}
+					root="quiz-atom"
+					index={quizAtoms.quizIndex}
+				>
+					{quizAtoms.quizType === 'personality' &&
+					quizAtoms.resultBuckets ? (
```

Comment:
This logic could result in a quiz with type 'personality' being rendered by the KnowledgeQuizAtom which we proabbly don't want. Maybe we can handle resultBuckets being undefined in the PersonalityQuizAtom itslef? Or, and this might be better, if it really is required we should make it so in the type system so we don't need to have this guard

Reply:
this is a good point, what do you think of removing the turnary and instead having something like this?
```
					{quizAtoms.quizType === 'personality' &&
						quizAtoms.resultBuckets && (
							<PersonalityQuizAtom
								id={quizAtoms.id}
								questions={quizAtoms.questions}
								resultBuckets={quizAtoms.resultBuckets}
							/>
						)}
					{quizAtoms.quizType === 'knowledge' && (
						<KnowledgeQuizAtom
							id={quizAtoms.id}
							questions={quizAtoms.questions}
						/>
					)}
```

I am reluctant to have to handled `quizAtoms.resultBuckets` within Quiz mainly because without it we do not have a way of displaying awnsers to users (thus rendering the quiz pointless).

Reply:
So that's better in that it prevents 'personality' quizzes being rendered as Knowledge quizzes. But it still opens up the possibility of null results when really we want to prevent this with typing.

It feel like resultBuckets should be a required non nullable list and we should not accept from CAPI anything outside of that. Does that sound right?

Reply:
> It feel like resultBuckets should be a required non nullable list and we should not accept from CAPI anything outside of that. Does that sound right?

the issue is that we have 1 quiz type `model.dotcomrendering.pageElements.QuizAtomBlockElement` that technically splits out into Personality Quiz types (that require `resultBuckets`) and Knowledge Quiz types that do not require that prop. Should this seperation be done in CAPI to avoid this possibility?

Reply:
From internal discussions there seems little appetite to split the `QuizAtomBlockElement` into 2 blocks. I will close this thread as a result